### ISSUE TITLE
Testing Data loss using MQTT + Azure

### ISSUE THREAD

**Issue by AnshumanFauzdar** [SwayattDrishtigochar/Sensor-Module#6]
## Python code

```
import csv
from datetime import datetime, timedelta

data_file = "data.csv"

with open(data_file, "r") as file:
    csv_reader = csv.reader(file)
    headers = next(csv_reader)

    # Initialize variables
    last_timestamp = None
    total_loss = 0
    line_num = 1

    for row in csv_reader:
        # Extract timestamp from the row
        timestamp_str = row[2].strip('timestamp:"}{')
        timestamp = datetime.strptime(timestamp_str, '%Y-%m-%d %H:%M:%S.%f')

        # Check if this is the first row or if the time difference between this row and the last row is less than or equal to 3 seconds
        if last_timestamp is None or (timestamp - last_timestamp) <= timedelta(seconds=3):
            last_timestamp = timestamp
        else:
            # Calculate the time difference between this row and the last row
            time_diff = (timestamp - last_timestamp).total_seconds()

            # Check if the time difference is more than 2 seconds and less than 3 seconds
            if 2 < time_diff < 3:
                print(f"Data loss at line {line_num}: {time_diff:.3f} seconds")
                total_loss += 1

        line_num += 1

print(f"Total data loss: {total_loss}")
```
## Azure cloud storage account :
- Container : csv
- Blob storage : https://sensormodule.blob.core.windows.net/csv/2023-05-17_16-15-17_test_data.csv

## Results
- 17-05-2023 : 0%

## CSV upload from pi [Node-red and MQTT broker]

```
import datetime
from azure.storage.blob import BlobServiceClient

# Azure Storage account connection string
connection_string = "AZURE_STORAGE_STRING"

# Local path of the CSV file on the Raspberry Pi
local_file_path = "data.csv"

# Name of the container in Azure Blob storage
container_name = "csv"

# Name for the uploaded CSV file in Azure Blob storage
timestamp = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")

blob_name = timestamp + "_test_data.csv"

# Create a BlobServiceClient object using the connection string
blob_service_client = BlobServiceClient.from_connection_string(connection_string)

# Get a reference to the container
container_client = blob_service_client.get_container_client(container_name)

# Upload the CSV file to Azure Blob storage
with open(local_file_path, "rb") as data:
    container_client.upload_blob(name=blob_name, data=data)
```
